app/src/main/java/com/android/sample/model/recipe/MealDBRecipesRepository.kt

```diff
@@ -0,0 +1,238 @@
+package com.android.sample.model.recipe
+
+import com.android.sample.resources.C.Tag.MAXIMUM_RECIPES_TO_FETCH_MEAL_DB
+import com.android.sample.resources.C.Tag.MAX_NB_OF_INGREDIENTS_IN_A_RECIPE_MEAL_DB
+import com.android.sample.resources.C.Tag.MEAL_DB_USER_AGENT
+import com.android.sample.resources.C.Tag.MEAL_DB_USER_AGENT_VALUE
+import java.io.IOException
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Repository for fetching recipes from the MealDB API.
+ *
+ * @property client The OkHttpClient used to make network requests.
+ */
+class MealDBRecipesRepository(private val client: OkHttpClient) : RecipesRepository {
+
+  private val mealDBUrl = "https://www.themealdb.com/api/json/v1/1/"
+
+  /**
+   * Fetches a specified number of random recipes from the MealDB API.
+   *
+   * @param json The JSON object to parse.
+   */
+  private fun parseMealDBJsonToRecipe(json: JSONObject): List<Recipe> {
+    val parsedListOfRecipes = mutableListOf<Recipe>()
+    val listOfRecipes = json.getJSONArray("meals")
+    for (i in 0 until listOfRecipes.length()) {
+      val meal = listOfRecipes.getJSONObject(i)
+      val idMeal = meal.getString("idMeal")
+      val strMeal = meal.getString("strMeal")
+      val strCategory = meal.getString("strCategory")
+      val strArea = meal.getString("strArea")
+      val strInstructions = meal.getString("strInstructions")
+      val strMealThumbUrl = meal.getString("strMealThumb")
+      val ingredientsAndMeasurements = mutableListOf<Pair<String, String>>()
+      var j = 1
+      while ((j <= MAX_NB_OF_INGREDIENTS_IN_A_RECIPE_MEAL_DB) &&
+          meal.optString("strIngredient$j", "").isNotEmpty()) {
+
+        val ingredient = meal.optString("strIngredient$j", "")
+        val measurement = meal.optString("strMeasure$j", "")
+        if (ingredient.isNotEmpty() && measurement.isNotEmpty()) {
+          ingredientsAndMeasurements.add(Pair(ingredient, measurement))
+        }
+        j++
+      }
+      parsedListOfRecipes.add(
+          Recipe(
+              idMeal,
+              strMeal,
+              strCategory,
+              strArea,
+              strInstructions,
+              strMealThumbUrl,
+              ingredientsAndMeasurements))
+    }
+    return parsedListOfRecipes
+  }
+
+  /**
+   * Parses the JSON object to a list of categories.
+   *
+   * @param json The JSON object to parse.
+   */
+  private fun parseMealDBJsonToCategory(json: JSONObject): List<String> {
+    val parsedListOfCategories = mutableListOf<String>()
+    val listOfCategories = json.getJSONArray("meals")
+    for (i in 0 until listOfCategories.length()) {
+      val category = listOfCategories.getJSONObject(i)
+      val strCategory = category.getString("strCategory")
+      parsedListOfCategories.add(strCategory)
```

Comment:
You could move your constant to the `resources/c` file

Reply:
Done